### DISCUSSION TITLE
Behaviour when using line tool

### DISCUSSION THREAD

**Discussion by Robinlovelace** [acteng/atip#50] [Q&A] [answered]
Sometimes when using the line tool without snapping red dots appear on the map without being joined. They disappear when you 'finish snapping' (shouldn't that be 'finish route'?) but are a unhelpful while you're in progress, as shown in the vid below.

https://user-images.githubusercontent.com/1825120/205684489-6cf0248a-5732-4306-b217-381ab9c800d7.mp4

**Comment by dabreegster** [comment, marked as answer]
It's indeed flaky, as your video shows. Not sure what's happening yet; I'll find a reliable repro and add logging to see if a waypoint is being treated internally as snapped or freeform. I suspect the former, which is why it'd be dropping some of them sometimes.
